storage/immunitycache/chunk.go

```diff
@@ -147,8 +147,8 @@ func (chunk *immunityChunk) removeOldestNoLock(numToRemove int) int {
 }
 
 func (chunk *immunityChunk) removeNoLock(element *list.Element) {
-	item := element.Value.(storage.CacheItem)
-	delete(chunk.items, string(item.GetKey()))
+	item := element.Value.(*cacheItem)
+	delete(chunk.items, item.key)
```

Comment:
check with ok the type assertion return if wrong. same for every type assertion.